File: Modul 6/C/PRAK604-2310817310005-MuhammadNurwahyudiAdhitama.c
#include <stdio.h>
#include <string.h>

void main() {
    char baris1[50],baris2[50];

    scanf(" %[^\n]", &baris1);

    scanf(" %[^\n]", &baris2);

    printf("\n");

    size_t SIZE_baris1 = strlen(baris1);
    size_t SIZE_baris2 = strlen(baris2);

    if(SIZE_baris1 == SIZE_baris2) {

        size_t i = 0;
        size_t j = 0;

        int JUMLAH_ASTERISK = 0;
        int JUMLAH_HASH = 0;

        while(i < SIZE_baris1 && j < SIZE_baris2) {

            if(baris1[i] == ' ' && baris2[j] == ' '){
                i++;
                j++;
                continue;
            }
            if(baris1[i] == baris2[j]) {
                printf("*");
                JUMLAH_ASTERISK++;
            }
            else {
                printf("#");
                JUMLAH_HASH++;
            }
            i++;
            j++;
        }

        printf("\n* = %d\n", JUMLAH_ASTERISK);
        printf("# = %d\n", JUMLAH_HASH);

        if (JUMLAH_ASTERISK >= JUMLAH_HASH) {
            printf("Pesan Asli");
        }
        else {
            printf("Pesan Palsu");
        }

    }

    else {
        printf("Panjang kalimat berbeda, pesan palsu");
    } 
}

File: Modul 6/Python/PRAK604-2310817310005-MuhammadNurwahyudiAdhitama.py
import numpy as np

baris1 = input("")
baris2 = input("")

print()

arr1 = np.array(baris1).reshape(1,1)
arr2 = np.array(baris2).reshape(1,1)

SIZE_baris1 = len(baris1)
SIZE_baris2 = len(baris2)

if (SIZE_baris1 == SIZE_baris2) :

    i = 0
    j = 0
    JUMLAH_ASTERISK = 0
    JUMLAH_HASH = 0

    while (i < SIZE_baris1 and j < SIZE_baris2) :
        
        if (baris1[i] == ' ' and baris2[j] == ' ') :
            i += 1
            j += 1
            continue

        if (baris1[i] == baris2[j]) :
            print("*", end = "")
            JUMLAH_ASTERISK += 1

        else :
            print("#", end = "")
            JUMLAH_HASH += 1

        i += 1
        j += 1

    print(f"\n* = {JUMLAH_ASTERISK}")
    print(f"# = {JUMLAH_HASH}")

    if (JUMLAH_ASTERISK >= JUMLAH_HASH) :
        print("Pesan Asli")
    
    else :
        print("Pesan Palsu")
        
else :
    print("Panjang kalimat berbeda, pesan palsu")